controllers/component_build_controller_pac.go

```diff
@@ -181,11 +345,101 @@ func (r *ComponentBuildReconciler) UndoPaCProvisionForComponent(ctx context.Cont
 	}
 
 	// Manage merge request for Pipelines as Code configuration removal
-	mrUrl, action, err := r.UnconfigureRepositoryForPaC(ctx, component, pacSecret.Data, webhookTargetUrl)
+	baseBranch, mrUrl, action, err := r.UnconfigureRepositoryForPaC(ctx, component, pacSecret.Data, webhookTargetUrl)
 	if err != nil {
 		log.Error(err, "failed to create merge request to remove Pipelines as Code configuration from Component source repository", l.Audit, "true")
 		return "", err
 	}
+
+	// check if more components are using same repo with PaC enabled for incomings removal from repository
```

Comment:
I don't understand. What's the reason TriggerPaCBuild handles incomings removal. What causes an incoming is removed from a Repository?

Reply:
I am cleaning up incomings, and secret if they aren't needed anymore

there can be also manual user changes to Repository messing up it